tests/mir-opt/jump_threading.rs

```diff
@@ -531,6 +531,22 @@ fn floats() -> u32 {
     if x == 0.0 { 0 } else { 1 }
 }
 
+fn not_int() -> i32 {
```

Comment:
Please rebase this into a commit *before* you made the changes, so we can see the diff; usually I don't care about separate tests for "before the fix" and "after the fix" but for misfiring optimizations, I want to see that it was actually miscompiling before.